client_test/directfund_integration_test.go

```diff
@@ -5,52 +5,68 @@ import (
 	"math/big"
 	"math/rand"
 	"testing"
+	"time"
 
 	"github.com/statechannels/go-nitro/channel/state/outcome"
 	"github.com/statechannels/go-nitro/client"
 	"github.com/statechannels/go-nitro/client/engine/chainservice"
 	"github.com/statechannels/go-nitro/client/engine/messageservice"
+	"github.com/statechannels/go-nitro/protocols"
 	"github.com/statechannels/go-nitro/protocols/directfund"
 	"github.com/statechannels/go-nitro/types"
 )
 
-func directlyFundALedgerChannel(t *testing.T, alpha client.Client, beta client.Client) {
-	// Set up an outcome that requires both participants to deposit
-	outcome := outcome.Exit{outcome.SingleAssetExit{
-		Allocations: outcome.Allocations{
-			outcome.Allocation{
-				Destination: types.AddressToDestination(*alpha.Address),
-				Amount:      big.NewInt(5),
-			},
-			outcome.Allocation{
-				Destination: types.AddressToDestination(*beta.Address),
-				Amount:      big.NewInt(5),
+func createDirectlyFundedChannels(t *testing.T, alpha client.Client, beta client.Client, amountOfChannels uint) []protocols.ObjectiveId {
+	ids := []protocols.ObjectiveId{}
+	for i := uint(0); i < amountOfChannels; i++ {
+
+		// Set up an outcome that requires both participants to deposit
+		outcome := outcome.Exit{outcome.SingleAssetExit{
+			Allocations: outcome.Allocations{
+				outcome.Allocation{
+					Destination: types.AddressToDestination(*alpha.Address),
+					Amount:      big.NewInt(5),
+				},
+				outcome.Allocation{
+					Destination: types.AddressToDestination(*beta.Address),
+					Amount:      big.NewInt(5),
+				},
 			},
-		},
-	}}
-	request := directfund.ObjectiveRequest{
-		MyAddress:         *alpha.Address,
-		CounterParty:      *beta.Address,
-		Outcome:           outcome,
-		AppDefinition:     types.Address{},
-		AppData:           types.Bytes{},
-		ChallengeDuration: big.NewInt(0),
-		Nonce:             rand.Int63(),
+		}}
+		request := directfund.ObjectiveRequest{
+			MyAddress:         *alpha.Address,
+			CounterParty:      *beta.Address,
+			Outcome:           outcome,
+			AppDefinition:     types.Address{},
+			AppData:           types.Bytes{},
+			ChallengeDuration: big.NewInt(0),
+			Nonce:             rand.Int63(),
+		}
+		id := alpha.CreateDirectChannel(request)
+		ids = append(ids, id)
 	}
-	id := alpha.CreateDirectChannel(request)
+	return ids
+}
+func directlyFundALedgerChannel(t *testing.T, alpha client.Client, beta client.Client) {
+	ids := createDirectlyFundedChannels(t, alpha, beta, 1)
+	id := ids[0]
 	waitTimeForCompletedObjectiveIds(t, &alpha, defaultTimeout, id)
 	waitTimeForCompletedObjectiveIds(t, &beta, defaultTimeout, id)
 }
+
+const MEAN_MESSAGE_DELAY = 100 * time.Millisecond
```

Comment:
How did you arrive at this number?

It seems to me as if what we really want is a system for generating random _orderings_ for messages that are to be dispatched. Using random delays is just one proxy for achieving that. Of course, random delays are very realistic and attractive to use in testing for that reason; but if the main symptom of message delays is message reordering, we might be smart to simply shortcut straight to reordering in our tests.

If we do stick with the random delay method, then it seems the condition we want to be in to have reordering (with high probability) is that the _variance_ in the delay is large. But if it gets too large we are needlessly slowing down the test. In your current system, the variance will increase if the mean increases.

Thinking about the distribution of delays again, it might make sense to specify a _maximum_ delay rather than a mean delay -- and that way, include delays of 0 in the distribution. Having a nonzero minimum delay doesn't make a heap of sense to me (if we are just trying to reorder messages). It just adds an unnecessary (albeit negligible, in all likelihood) lag to the test.